src/server/world.zig

```diff
@@ -1061,12 +1065,21 @@ pub const ServerWorld = struct { // MARK: ServerWorld
 			defer if(ch != baseChunk) {
 				ch.decreaseRefCount();
 			};
+
 			ch.mutex.lock();
 			defer ch.mutex.unlock();
+
 			var neighborBlock = ch.getBlock(nx & chunk.chunkMask, ny & chunk.chunkMask, nz & chunk.chunkMask);
 			if(neighborBlock.mode().dependsOnNeighbors) {
 				if(neighborBlock.mode().updateData(&neighborBlock, neighbor.reverse(), newBlock)) {
 					ch.updateBlockAndSetChanged(nx & chunk.chunkMask, ny & chunk.chunkMask, nz & chunk.chunkMask, neighborBlock);
+					const wnx = wx + neighbor.relX();
+					const wny = wy + neighbor.relY();
+					const wnz = wz + neighbor.relZ();
+					for(userList) |user| {
+						std.log.debug("Neighbor update ({d:.2} {d:.2} {d:.2}) {s}:{d}", .{wnx, wny, wnz, neighborBlock.id(), neighborBlock.data});
+						main.network.Protocols.blockUpdate.send(user.conn, wnx, wny, wnz, neighborBlock);
```

Comment:
Could you instead batch it and only send it once in bulk in the end? We shouldn't do 7 network operations (each locking the network mutex) per user here, and the protocol overhead is also a thing (even if it's just 2 bytes here).